core/src/main/java/edu/wpi/grip/core/sources/NetworkTableEntrySource.java

```diff
@@ -0,0 +1,148 @@
+  }
+
+  public enum Types {
+    BOOLEAN, NUMBER, STRING;
+
+    @Override
+    public String toString() {
+      return CaseFormat.UPPER_UNDERSCORE.to(CaseFormat.UPPER_CAMEL, name());
+    }
+
+  }
+
+  @AssistedInject
+  NetworkTableEntrySource(
+      EventBus eventBus,
+      ExceptionWitness.Factory exceptionWitnessFactory,
+      OutputSocket.Factory osf,
+      @Named("ntManager") MapNetworkReceiverFactory networkReceiverFactory,
+      @Assisted Properties properties) {
+    this(eventBus,
+        exceptionWitnessFactory,
+        osf,
+        networkReceiverFactory,
+        properties.getProperty(PATH_PROPERTY),
+        Types.valueOf(properties.getProperty(TYPE_PROPERTY)));
+  }
+
+  @AssistedInject
+  NetworkTableEntrySource(
+      EventBus eventBus,
+      ExceptionWitness.Factory exceptionWitnessFactory,
+      OutputSocket.Factory osf,
+      @Named("ntManager") MapNetworkReceiverFactory networkReceiverFactory,
+      @Assisted String path,
+      @Assisted Types type) {
+    super(exceptionWitnessFactory);
+    this.eventBus = eventBus;
+    this.path = path;
+    this.type = type;
+    networkReceiver = networkReceiverFactory.create(path);
+    output = osf.create(createOutputSocket(type));
+  }
+
+  @Override
+  public String getName() {
+    return path;
+  }
+
+  @Override
+  protected List<OutputSocket> createOutputSockets() {
+    return ImmutableList.of(
+        output
+    );
+  }
+
+  @Override
+  protected boolean updateOutputSockets() {
+    try {
+      output.setValue(networkReceiver.getValue());
+    } catch (ClassCastException ex) {
+      getExceptionWitness().flagException(ex, getName() + " is not of type "
+          + output.getSocketHint().getTypeLabel());
+      return false;
+    }
+    return true;
+  }
+
+  @Override
+  public Properties getProperties() {
+    Properties properties = new Properties();
+    properties.setProperty(PATH_PROPERTY, path);
+    properties.setProperty(TYPE_PROPERTY, type.toString().toUpperCase());
+    return properties;
+  }
+
+  @Override
+  public void initialize() {
+    networkReceiver.addListener(o -> eventBus.post(new SourceHasPendingUpdateEvent(this)));
+  }
+
+  @Subscribe
+  public void onSourceRemovedEvent(SourceRemovedEvent event) {
+    if (event.getSource() == this) {
+      networkReceiver.close();
+    }
+  }
+
+  /**
+   * Create a SocketHint from the given type.
+   *
+   * @param type The type of SocketHint to create
+   */
+  private static SocketHint createOutputSocket(Types type) {
+    switch (type) {
+      case BOOLEAN:
+        return SocketHints.Outputs.createBooleanSocketHint(Types.BOOLEAN.toString(), false);
+      case NUMBER:
+        return SocketHints.Outputs.createNumberSocketHint(Types.NUMBER.toString(), 0.0);
+      case STRING:
+        return SocketHints.Outputs.createStringSocketHint(Types.STRING.toString(), "");
+      default:
+        throw new IllegalArgumentException(type + " is an invalid NetworkTable source type");
```

Comment:
It would be better to have a `private final Supplier<SocketHInt>` field that can get accessed and invoked.